### DISCUSSION TITLE
Twig creates wrong viewport width on mobile and tablet

### DISCUSSION THREAD

**Discussion by snorky22** [symfony/symfony#52225] [Q&A]
### Symfony version(s) affected

6.3.5

### Description

I have a problem with twig, seemingly, specifically with device viewport. I have installed webpack Encore.

I am testing some Symfony code on mobile devices, and I add some code that returns the viewport width.

When I compute the raw code for my web page into the variable `$html` and my controller returns `new Response($html)`, I check that the viewport of my iPhone is 393px, which is normal. On my tablet, the viewport is 744px, which is also normal.

When I use twig, the twig body simply returns the window width, computed the same way as previously. I do not even input extra code. Then I get a viewport of 980px both on my iPhone and on my tablet.

I have tested this on my local network and also from a remote server. I use actual devices for all my tests.

This has very serious consequences when it comes to responsive design. Actually, I have done all of these tests because I could not set my font sizes correctly on my phone and tablet when testing my Symfony app.

All of this has been done on a fresh install of Symfony+Encore.

Can you help me?

Tia,
François

### Controller raw code method
1. Generate dummy header and footer files
2.  The controller code is
```
$header = file_get_contents('header.html');
$footer = file_get_contents('footer.html');
$html = $header;
$html .= '<p id="windowWidth"></p>';
$html .= '<script>let ww = window.innerWidth;let el = document.getElementById("windowWidth");el.innerHTML = "Window width is "+ww+"px<br>and visual viewport is "+window.visualViewport.width;</script>';
$html.= $footer;
return new Response($html);
```

This returns standard values for the viewport widths: 393px for my iPhone and 744px for my iPad.
### Controller twig code method
The controller returns
```
return $this->render('tester/index.html.twig');
```
with the following twig file
```
 <p id="windowWidth"></p>
<script>
        let ww = window.innerWidth;
        let el = document.getElementById("windowWidth");
        el.innerHTML = "Window width is "+ww+"px<br>and visual viewport is "+window.visualViewport.width;
 </script>
```
which uses the same javascript code as previously.

This method returns a viewport width of 980px both for my iPhone and my iPad.

**Comment by MatTheCat** [comment]
I guess your `$header` adds something like `<meta name="viewport" content="width=device-width">`. If you don't include it in your Twig template the browser will set its own width, the default being 980px.

The issue would be your HTML, not Twig (neither Symfony).

**Reply by snorky22** [reply]
Thanks a lot!
 As I said, I did a fresh installation of Symfony+Encore, and the `base.html.twig` did not contain this meta tag (which is surprising). Once I added it to the base template, everything worked normally.

Thanks again for your help!

**Reply by snorky22** [reply]
I have just re-installed Symfony with
```
symfony new my_project_directory --version="6.3.*" --webapp
```
and the `base.html.twig` does not include the viewport meta tag.

Good thing to know.
